### ISSUE TITLE
regex for sanitising is invalid

### ISSUE THREAD

**Issue by bgotink** [kevin-smets/caml#4]
Regex `/[",'].*\..*[",']:/` matches stuff like the following yaml file incorrectly:

``` yaml
'"this is": a': test
```

In yaml:

``` js
> require('js-yaml').safeLoad(require('fs').readFileSync('test.yml'))
{ '"this.is": a': 'test' }
```

In caml:

```
$ caml -f test

/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/scanner.js:504
            throw new exports.ScannerError(null, null, 'mapping values are not allowed here', this.get_mark());
            ^
mapping values are not allowed here
  on line 1, column 15
    at ScannerError.YAMLError [as constructor] (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/errors.js:70:46)
    at ScannerError.MarkedYAMLError [as constructor] (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/errors.js:90:45)
    at new ScannerError (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/scanner.js:23:49)
    at Constructor.__dirname.Scanner.Scanner.fetch_value (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/scanner.js:504:19)
    at Constructor.__dirname.Scanner.Scanner.fetch_more_tokens (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/scanner.js:214:21)
    at Constructor.__dirname.Scanner.Scanner.check_token (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/scanner.js:115:14)
    at Constructor.__dirname.Parser.Parser.parse_block_mapping_key (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/parser.js:421:16)
    at Constructor.__dirname.Parser.Parser.check_event (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/parser.js:61:48)
    at Constructor.__dirname.Composer.Composer.compose_mapping_node (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/composer.js:143:20)
    at Constructor.__dirname.Composer.Composer.compose_node (/usr/local/lib/node_modules/caml/node_modules/yaml-js/lib/composer.js:91:21)
```

**Comment by kevin-smets** [comment]
Why would you create such a weird identifier :P. But yes, point taken. Caml is being reworked from the ground up to properly parse such exotic constructions (branch parser).

**Comment by asgoth** [comment]
To hijack the issue: sanitize is too eager. Using comments after a property causes the property to dissapear, e.g.

```
path:
  to:
    some: "value" # some explanation what this does
    other: "value"
```

Results in

```
{
  path: {
    to: {
      other: "value"
    }
  }
}
```

Property `some` is gone.

**Comment by kevin-smets** [comment]
Yes, the sanitizer is overly aggressive. For now a workaround is to make sure your comments are on a dedicated line.

**Comment by kevin-smets** [comment]
I've parked the parser for now, without proper documentation of yaml-js (https://github.com/connec/yaml-js/issues/12) I simply do not have the time to rework this from the ground up for now.

@asgoth could you log this into a new issue? This might still be tackled in the current regex based setup.

**Comment by asgoth** [comment]
Just use this one, no?

Edit: or you mean about the comments?

**Comment by bgotink** [comment]
I believe the original issue here was already solved in #7 (by explicitly disallowing nested quotes in object keys.

**Comment by kevin-smets** [comment]
Alright, will close this one, I'll rise another issue from this one's ashes. Closed by #7
